stdlib/3/multiprocessing/sharedctypes.pyi

```diff
@@ -0,0 +1,133 @@
+import ctypes
+from typing import overload, Any, ContextManager, Generic, Optional, Sequence, Type, TypeVar, Union
+
+from . import synchronize
+from . import context
+
+_LockLike = Union[synchronize.Lock, synchronize.RLock]
+_T = TypeVar("_T", bound=ctypes._CData)
+_U = TypeVar("_U")
+
+# TODO: typecode_or_type should be in array.typecode
+@overload
+def RawValue(typecode_or_type: str, *args: Any) -> ctypes._SimpleCData: ...
+@overload
+def RawValue(typecode_or_type: Type[_T], *args: Any) -> _T: ...
+@overload
+def RawArray(typecode_or_type: str, size_or_initializer: Union[int, Sequence[Any]]) -> ctypes.Array[ctypes._SimpleCData]: ...
+@overload
+def RawArray(typecode_or_type: Type[_T], size_or_initializer: Union[int, Sequence[Any]]) -> ctypes.Array[_T]: ...
+
+# Definitely lock case (can guarantee it's synchronized)
+@overload
+def Value(typecode_or_type: str, *args: Any, ctx: Optional[context.BaseContext] = ...) -> Synchronized: ...
+@overload
+def Value(
+    typecode_or_type: Type[ctypes._SimpleCData[_U]], *args: Any, ctx: Optional[context.BaseContext] = ...
+) -> Synchronized[_U]: ...
+@overload
+def Value(
+    typecode_or_type: Type[ctypes.Array[ctypes.c_char]], *args: Any, ctx: Optional[context.BaseContext] = ...
+) -> SynchronizedString: ...
+@overload
+def Value(
+    typecode_or_type: Type[ctypes.Array[_T]], *args: Any, ctx: Optional[context.BaseContext] = ...
+) -> SynchronizedArray[_T]: ...
+
+# Possibly locked case (don't know whether it's synchronized)
+@overload
+def Value(
+    typecode_or_type: str, *args: Any, lock: bool, ctx: Optional[context.BaseContext] = ...
+) -> Union[ctypes._SimpleCData, Synchronized]: ...
+@overload
+def Value(
+    typecode_or_type: Type[ctypes._SimpleCData[_U]], *args: Any, lock: bool, ctx: Optional[context.BaseContext] = ...
+) -> Union[ctypes._SimpleCData[_U], Synchronized[_U]]: ...
+@overload
+def Value(
+    typecode_or_type: Type[ctypes.Array[ctypes.c_char]], *args: Any, lock: bool, ctx: Optional[context.BaseContext] = ...
+) -> Union[ctypes.Array[ctypes.c_char], SynchronizedString]: ...
+@overload
+def Value(
+    typecode_or_type: Type[ctypes.Array[_T]], *args: Any, lock: bool, ctx: Optional[context.BaseContext] = ...
+) -> Union[ctypes.Array[_T], SynchronizedArray[_T]]: ...
+
+# Dynamically generates class, just type as Any
+@overload
+def Value(typecode_or_type: _T, *args: Any, lock: bool = ..., ctx: Optional[context.BaseContext]) -> Any: ...
```

Comment:
It looks like `lock` can also be None or some sort of Lock object with an `acquire` method. There don't seem to be any overloads accepting that.

Reply:
I'm going to use `_Optional[_LockLike]` for an overload here then.